server/utils.go

```diff
@@ -173,3 +175,106 @@ func fixGithubNotificationSubjectURL(url string) string {
 func fullNameFromOwnerAndRepo(owner, repo string) string {
 	return fmt.Sprintf("%s/%s", owner, repo)
 }
+
+// filter lines in a string from start to end
+func filterLines(s string, start, end int) string {
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	var buf strings.Builder
+	i := 1
+	for scanner.Scan() {
+		if i < start {
+			i++
+			continue
+		}
+		if i > end {
+			break
+		}
+		buf.Write(scanner.Bytes())
+		buf.WriteByte(byte('\n'))
+		i++
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Println(err)
+	}
+
+	return buf.String()
+}
+
+// getLineNumbers return the start and end lines from an anchor tag
+// of a github permalink.
+func getLineNumbers(s string) (start, end int) {
+	// split till -
+	parts := strings.Split(s, "-")
+
+	if len(parts) > 2 {
+		return -1, -1
+	}
+
+	switch len(parts) {
+	case 1: // just a single line
+		l := getLine(parts[0])
+		if l == -1 {
+			return -1, -1
+		}
+		if l < 3 {
+			return 0, l + 3
+		}
+		return l - 3, l + 3
+	case 2: // a line rage
+		start := getLine(parts[0])
```

Comment:
We should add a setting to limit the amount of lines we want to show at a time like @aaronrothschild suggested

Reply:
Done